snakemake/subworkflow.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from snakemake.logging import logger
+from snakemake.exceptions import WorkflowError
+
+
+def subworkflow(
+    path,
+    basedir,
+    input,
+    output,
+    params,
+    wildcards,
+    threads,
+    resources,
+    log,
+    config,
+    rulename,
+    conda_env,
+    container_img,
+    singularity_args,
+    env_modules,
+    bench_record,
+    jobid,
+    bench_iteration,
+    cleanup_scripts,
+    shadow_dir,
+    workflow,
+):
+    # extract (and normalize) parameters
+    basedir = os.path.normpath(basedir)
+    sub_workdir = os.path.normpath(params.get("workdir", f"subworkflow_{rulename}"))
+    sub_configfile = params.get("configfile", "")
```

Comment:
```suggestion
    sub_configfile = params.get("configfile")
```
don't use an empty string when you really want `None`